### PR TITLE
[DPEDE-1415] Added icon property to dropdown

### PR DESCRIPTION
**Pull request by wojciechwlodarczyklumen**
https://ctl.atlassian.net/browse/DPEDE-1415

### CODE REVIEW

**Comment by ASanchLumen** [comment]
Hi @wojciechwlodarczyklumen! The branch has conflicts, please resolve!

File: src/website/views/components/dropdown/_examples.pug [hunk 1]
```diff
@@ -775,6 +775,100 @@ p.-text
         <!-- Include your custom logic here to retain dropdown selection -->
       </script>
 
+.-portal--hide
+  h3 Icon
+  p.-text
+    | There is available property <code>icon</code> which allows to add an icon as the dropdown trigger.
```
Comment (ASanchLumen) [inline src/website/views/components/dropdown/_examples.pug:781]: Hi! Just some grammar stuff:

`There is an available property, <code>icon</code>, which allows adding an icon as the dropdown trigger.`

**Comment by ASanchLumen** [comment]
Should this feature have visual tests @josecarlosgonzalezv?

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

File: src/documentation/pages/components-wip/dropdown/examples/_icon_trigger.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<template lang="pug">
+<ComponentExample title="Icon" id="icon_trigger" :tabs="exampleTabs">
+  template(#example-description)
+    p.-text
+      | There is an available property, <code>icon</code>, which allows adding an icon as the dropdown trigger.
+  template(#example)
+    .chi-dropdown
+      button#example__dropdown-icon.chi-button.chi-dropdown__trigger.chi-dropdown__icon(aria-label="More options" data-tooltip="More options")
```
Comment (ASanchLumen) [inline src/documentation/pages/components-wip/dropdown/examples/_icon_trigger.vue:8]: In vue docs we can use the web component. For portal theme, in the website documentation it's being used  `variant=flat`. In this docs you can access the selected theme with `selectedTheme = useSelectedTheme();` and use the right variant: `:variant="selectedTheme === 'portal' ? 'flat' : ''".

Code snippets should also reflect this change.

File: src/website/views/components/dropdown/_examples.pug [hunk 3]
```diff
@@ -775,6 +775,100 @@ p.-text
...
+
+        <script>
+          chi.dropdown(document.getElementById('example__dropdown-icon'));
+          chi.tooltip(document.querySelectorAll('[data-tooltip]'));
+        </script>
+
+.-portal--show
+  h3 Icon
+  p.-text
+    | There is available property <code>icon</code> which allows to add an icon as the dropdown trigger.
```
Comment (ASanchLumen) [inline src/website/views/components/dropdown/_examples.pug:828]:
```suggestion
    | There is an available property, <code>icon</code>, which allows adding an icon as the dropdown trigger.
```

File: src/documentation/pages/components-wip/dropdown/examples/_retain_selection.vue [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      },
+      {
+        active: true,
+        id: 'htmlblueprint',
+        label: 'HTML Blueprint',
+      },
+    ];
+
+    const codeSnippets = {
+      webcomponent: ``,
```
Comment (ASanchLumen) [inline src/documentation/pages/components-wip/dropdown/examples/_retain_selection.vue:51]: Codesnippet missing for web component

File: src/documentation/pages/components-wip/dropdown/examples/_icon_trigger.vue [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      },
+      {
+        active: true,
+        id: 'htmlblueprint',
+        label: 'HTML Blueprint',
+      },
+    ];
+
+    const codeSnippets = {
+      webcomponent: ``,
```
Comment (ASanchLumen) [inline src/documentation/pages/components-wip/dropdown/examples/_icon_trigger.vue:43]: codeSnippet missing for web component!

**Review by ASanchLumen — CHANGES_REQUESTED** [review]
Update docs, otherwise ok!

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Review by ASanchLumen — APPROVED** [review]

**Pull request merged**